File: src/main/java/net/atos/client/zgw/ztc/model/AbstractZTCListParameters.java
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2023-2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.client.zgw.ztc.model;

import jakarta.ws.rs.QueryParam;

/**
 *
 */
public abstract class AbstractZTCListParameters {

  /*
   * Filter objects depending on their concept status
   */
  private ObjectStatusFilter status;

  @QueryParam("status")
  public String getStatus() {
    return status != null ? status.toValue() : null;
  }

  public void setStatus(final ObjectStatusFilter status) {
    this.status = status;
  }
}

File: src/main/kotlin/net/atos/client/zgw/ztc/model/AbstractZTCListParameters.kt
/*
 * SPDX-FileCopyrightText: 2021 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.client.zgw.ztc.model

import jakarta.ws.rs.QueryParam

/**
 *
 */
abstract class AbstractZTCListParameters {
    /*
     * Filter objects depending on their concept status
     */
    private var status: ObjectStatusFilter? = null

    @QueryParam("status")
    fun getStatus(): String? {
        return if (status != null) status!!.toValue() else null
    }

    fun setStatus(status: ObjectStatusFilter?) {
        this.status = status
    }
}
